### PR TITLE
Upgrade py-spy

### PR DESCRIPTION
**Pull request by Jongy**
## Description
Includes a not yet merged fix from https://github.com/benfred/py-spy/pull/407

## Motivation and Context
Fixes a "bug" in py-spy and prevented us from profiling certain Python builds.
See explanation in the mentioned PR.

## How Has This Been Tested?
CI

## Checklist:
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have updated the relevant documentation.
- [ ] I have added tests for new logic.

### CODE REVIEW

**Comment by Jongy** [comment]
Tested gProfiler built from this branch on Python 2.7.6 (`python` package of the `ubuntu:14.04` image), profiling works, and it didn't work before (failed with the error as described in https://github.com/benfred/py-spy/pull/407)

**Review by guybortnikov — APPROVED** [review]

**Comment by guybortnikov** [comment]
Wait for CI to pass before merging

**Comment by Jongy** [comment]
Random failure, not something I wish to debug in this PR. I'll merge it in the morning, not the best idea to release a new version at 2AM

**Pull request merged**
